tests/test_dependency_cache.py

```diff
@@ -29,7 +29,7 @@ async def get_sub_counter(
 
 @app.get("/sub-counter-no-cache/")
 async def get_sub_counter_no_cache(
-    subcount: int = Depends(super_dep),
+    subcount: int = Depends(dep_counter),
```

Comment:
Here I had to change this to the topmost dep, because one level lower already resolves just a tiny bit later and in that case this test is not valid anymore.